docs/specs/config.md

```diff
@@ -11,24 +11,45 @@
 5. Naming:
    1. Use `camelCase`.
    2. Use plural form for array value.
-   3. Use singular form for str value.
-   4. Use plural form for an object that key is not fixed. e.g. `values` in `metadata`.
+   3. Use singular form for str/object value.
+   4. Use plural form for an object that key is not fixed. e.g. `values` in `fileMetadata`.
    5. Use plural form for an object that key is fixed. e.g. `contribution`.
    6. Use singular form if type is dynamic. e.g. `path`, although it can be a list.
+   7. Use verb's original form, instead of simple present form.
 6. If some settings need to be applied to file(s), use filename of foldername as the key, not glob pattern, to avoid `**.md` everywhere.
 
-## Major Changes
+## Key Points
 1. All configs that doesn't affect build output and only consumed by OPS service side will not be defined here. They can be placed under a single object, named with `service_config`. Like: `need_preview_pull_request`, `notification_subscribers`.
    Steps to integrate new config:
    1. Replace `docfx.json`s with `docfx.yml`.
    2. OPS consumes `docfx.yml` to build output, use `.openpublishing.publish.config.json` to fetch service side config.
    3. Move remaining configs `.openpublishing.publish.config.json` to server side config, then remove it.
-2. Config is the source of truth when building. If the actual source of truth is in DHS, like `product_name` or `docset_name`, config should be updated on DHS change, and validated before each build.
+2. Config is the source of truth when building. If the actual source of truth is in DHS, like `product_name` or `docset_name` in v2, config should be updated on DHS change, and validated before each build.
 2. Abandon `docset` and `group` level config. All configs is applied to all files or a set of files matching certain glob pattern.
 3. Distinguish config from metadata:
    * Metadata: part of content, can be consumed by template. In general, it's passed through from input to template. DocFX core needn't consume it.
    * Config: not content, can be consumed by DocFX
-4. Merge GlobalMetadata and FileMetadata.
+4. Many configs share the pattern "I want to applies a spcified config to some files", like `fileMetadata`, `monikerRange`. We name it **file level config**. All these configs should use this pattern:
+   1. Object form: It is a simple form that supports folder and file level config:
```

Comment:
folder only?

Reply:
Tend to also support file. e.g. `routes` cannot support glob pattern, so the key should support file in this case. Other config can also follow this.